### ISSUE TITLE
[Enhancement]: Make segment small index configurable

### ISSUE THREAD

**Issue by longjiquan** [milvus-io/milvus#16924]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

I want to make segment small index configurable.

### Why is this needed?

We're not sure that segment small index will make search performance better. Sometimes maybe bruteforce search is faster. So hard code to enable segment small index is not convinient.

**Comment by longjiquan** [comment]
- 1, When small index is disable, search on growing segment will use bruteforce scan on all data chunks;
- 2, When small index is enable, related parameters can also be configurable:
   - a, For simplicity, we just plan to support `IVF_FLAT` as our segment small index;
   - b, Index parameters such as `nlist` can be configurable;
   - c, Search parameters on these segment small indexes can be configurable;

**Comment by xiaofan-luan** [comment]
lgtm, we just want to disable small index for now and need to add more test on top of it

**Comment by xige-16** [comment]
search params for small index is not necessarily configurable, it should be executed according to the search params in search req. such  as nprobe

**Comment by longjiquan** [comment]
Why should users pass `nprobe` when they'll search on these growing segments? @xige-16

**Comment by longjiquan** [comment]
It's too complicated that we maintain so many small-index-related parameters now. Maybe we can choose to enable it after we made a benchmark between segment small index and bruteforce search.

**Comment by xiaofan-luan** [comment]
> It's too complicated that we maintain so many small-index-related parameters now. Maybe we can choose to enable it after we made a benchmark between segment small index and bruteforce search.

Agree, we don't even have idea about the performance boost of enable small index~
